frontend/public/components/modals/configure-cluster-upstream-modal.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+
+import { ClusterVersionModel } from '../../models';
+import { ClusterVersionKind, k8sPatch } from '../../module/k8s';
+import {
+  ModalBody,
+  ModalComponentProps,
+  ModalSubmitFooter,
+  ModalTitle,
+  createModalLauncher,
+} from '../factory/modal';
+import { ExternalLink, HandlePromiseProps, withHandlePromise } from '../utils';
+import { useTranslation } from 'react-i18next';
+import { TFunction } from 'i18next';
+import { RadioInput } from '../radio';
+import {
+  CLUSTER_VERSION_DEFAULT_UPSTREAM_SERVER_URL,
+  CLUSTER_VERSION_CUSTOM_UPSTREAM_SERVER_URL_PLACEHOLDER,
+} from '@console/shared/src/constants';
+import { TextInput } from '@patternfly/react-core';
+
+export const ConfigureClusterUpstreamModal = withHandlePromise(
+  (props: ConfigureClusterUpstreamModalProps) => {
+    const { cv, handlePromise, close } = props;
+    const currentUpstream = cv?.spec?.upstream;
+
+    const [selectedRadio, setSelectedRadio] = React.useState(
+      currentUpstream ? 'custom' : 'default',
+    );
+    const [customUrl, setCustomUrl] = React.useState(currentUpstream);
+    const [invalidCustomUrl, setInvalidCustomUrl] = React.useState(false);
+
+    const submit: React.FormEventHandler<HTMLFormElement> = (e) => {
+      e.preventDefault();
+      if (selectedRadio === 'custom') {
+        if (!customUrl) {
+          setInvalidCustomUrl(true);
+          return;
+        } else if (customUrl === currentUpstream) {
+          return handlePromise(Promise.resolve(), close);
```

Comment:
I'm not sure this is right. If I previously had a different custom value and then manually typed the default, would the `ClusterVersion` resource be updated?

Reply:
@spadgett if you had a previous custom value, IE `https://openshift-release.apps.ci.l2s4.p1.openshiftapps.com/graph`, then manually typed in `https://api.openshift.com/api/upgrades_info/v1/graph` in the custom input, then yes, this will work (tested on my dev system). Only selected the `Default` radio will unset the upstream value.

Reply:
> Are you saying you want the UI to auto-detect if you typed in the exact value for the default configuration and unset the upstream accordingly?

Sorry, I misread. I thought you were checking current URL against the default URL (not the previous value). Disregard.